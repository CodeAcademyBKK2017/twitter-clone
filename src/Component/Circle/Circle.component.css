.circle{
    display: flex;
    width: 35px;
    height: 35px;
    border-radius: 50%;
    justify-content: center;
    align-items: center;
}

.circleColorBlue{
    background-color: #1e90ff;
}

.circleColor{
    border: 2px solid rgb(220, 220, 220);
}

.boxCircle{
    display: flex;
    width: 60px;
    height: 60px;
    justify-content: center;
    align-items: center;
}

.iconColorWhite {
    color: rgb(255, 255, 255);
}


