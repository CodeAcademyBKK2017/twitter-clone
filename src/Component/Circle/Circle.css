.Circle {
    display:flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    height: 35px;
    width: 35px;
    color:#80A8CC;
    border: 1px solid red
  } 