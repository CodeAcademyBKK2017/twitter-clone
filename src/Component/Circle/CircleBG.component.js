import './Circle.component.css';
import React, {Component} from 'react';
import {Icon} from 'react-fa';

class CircleBG extends Component {
  render () {
    return (
      <div className='Box-Circle'>
        <div className='Circle Circle-Color-Blue'><Icon className='Icon-Color-White' name={`${this.props.iconName}`}/></div>
      </div>
    );
  }
}

export default CircleBG;
