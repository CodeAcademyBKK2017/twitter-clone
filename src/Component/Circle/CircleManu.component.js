import inputStyles from './Circle.component.css';
import React, {Component} from 'react';
import Icon from 'react-ionicons';
import PropTypes from 'prop-types';
import className from 'classname';

class CircleManu extends Component {
  render () {
    return (
      <div className={className(inputStyles.boxCircleHide, inputStyles.boxCircle)}>
        <div className={className(inputStyles.circle, inputStyles.circleColorBlue)}>
          <Icon className={inputStyles.iconColorWhite} color={`${this.props.color}`} icon={`${this.props.iconName}`}/>
        </div>
      </div>
    );
  }
}

CircleManu.propTypes = {
  iconName: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired
};

export default CircleManu;
