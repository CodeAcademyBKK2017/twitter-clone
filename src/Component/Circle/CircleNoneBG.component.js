import inputStyles from './Circle.component.css';
import React, {Component} from 'react';
import {Icon} from 'react-fa';
import PropTypes from 'prop-types';
import className from 'classname';

class CircleNoneBG extends Component {
  render () {
    return (
      <div className={inputStyles.boxCircle}>
        <div className={className(inputStyles.circle, inputStyles.circleColor)}>
          <Icon name={`${this.props.iconName}`}/>
        </div>
      </div>
    );
  }
}

CircleNoneBG.propTypes = {
  iconName: PropTypes.string.isRequired
};

export default CircleNoneBG;
