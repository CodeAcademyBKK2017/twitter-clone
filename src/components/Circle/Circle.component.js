import CircleStyles from  './Circle.css';
import React, {Component} from 'react';
import Icon from 'react-ionicons';
import PropTypes from 'prop-types';

class Circle extends Component {

  render () {
    const {iconName, opaque, children} = this.props;
    const iconColor = opaque ? 'white' : '#605e58';
    const element = children || <Icon icon={iconName} color={iconColor} fontSize="20px"/>;
   
    return (
      <div className={`${CircleStyles.circle} ${opaque ? CircleStyles.circleOpaque : ''}`}>
        {element}
      </div>
    );
  }
}

Circle.propTypes = {
  iconName: PropTypes.string,
  children: PropTypes.node,
  opaque: PropTypes.bool
};

Circle.defaultTypes = {
  opaque: false
};

export default Circle;
