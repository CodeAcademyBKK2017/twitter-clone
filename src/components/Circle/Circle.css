.circle {
    width: 35px;
    height: 35px;
    border: 1px solid #ccc;
    border-radius: 35px;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-left: 10px;
}
.circle:hover {
    background-color: gray;
}

/* active: change to grey */
.circle:active {
    background-color: gray;
}

.background_sky{
    border: 1px solid #80A8CC;
    color: white;
    background-color: #80A8CC;
}

/*

height: 100%;
    display: flex;
    align-items: center;
    padding: 0 10px 0 10px;
    border-bottom: 2px solid red;

*/