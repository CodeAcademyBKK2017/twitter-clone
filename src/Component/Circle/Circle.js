import './Circle.css';
import React, {Component} from 'react';
import {Icon} from 'react-fa';

class Circle extends Component {
  render () {
      const {name} = this.props;
    return (     
        <div className='Circle'>
            <Icon name={name}/>
        </div>       
    );
  }
}

export default Circle;
