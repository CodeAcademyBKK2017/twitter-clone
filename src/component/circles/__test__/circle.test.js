import Circle from '../circle.component';
import React from 'react';
import renderer from 'react-test-renderer';
import {Icon} from 'react-fa';

describe('Circle component', () => {
  it('Circle: renders correctly', () => {
    const tree = renderer.create(<Circle><Icon name='bell-o'/></Circle>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});