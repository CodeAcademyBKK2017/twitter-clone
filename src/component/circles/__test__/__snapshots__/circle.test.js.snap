// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component Circle: renders correctly 1`] = `
<div
  className="sperateBTN isFlex Center"
>
  <div
    className="cirBTN isFlex Center"
  />
</div>
`;
